Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -108,59 +108,20 @@ public class PBXProjGenerator {
 
         pbxProj.rootObject = pbxProject
 
-        for target in project.targets {
-            let targetObject: PBXTarget
-
-            if target.isLegacy {
-                targetObject = PBXLegacyTarget(
-                    name: target.name,
-                    buildToolPath: target.legacy?.toolPath,
-                    buildArgumentsString: target.legacy?.arguments,
-                    passBuildSettingsInEnvironment: target.legacy?.passSettings ?? false,
-                    buildWorkingDirectory: target.legacy?.workingDirectory,
-                    buildPhases: []
-                )
-            } else {
-                targetObject = PBXNativeTarget(name: target.name, buildPhases: [])
-            }
-
-            targetObjects[target.name] = addObject(targetObject)
-
-            var explicitFileType: String?
-            var lastKnownFileType: String?
-            let fileType = Xcode.fileType(path: Path(target.filename))
-            if target.platform == .macOS || target.platform == .watchOS || target.type == .framework {
-                explicitFileType = fileType
-            } else {
-                lastKnownFileType = fileType
-            }
-
-            if !target.isLegacy {
-                let fileReference = addObject(
-                    PBXFileReference(
-                        sourceTree: .buildProductsDir,
-                        explicitFileType: explicitFileType,
-                        lastKnownFileType: lastKnownFileType,
-                        path: target.filename,
-                        includeInIndex: false
-                    ),
-                    context: target.name
-                )
-
-                targetFileReferences[target.name] = fileReference
-            }
+        let generatedTargetResults = self.generateTargetInfo(targets: self.project.targets, aggregateTargets: self.project.aggregateTargets)
```

Comment:
The motivation to move this logic to a different function was to break down the `generate` function into smaller functions.

And then the larger goal would be, to able to test out the `generateTargetInfo` function in isolation. But that requires marking the `generateTargetInfo` function as a public function so that it can be accessed in tests. 
It seems we could make this a pure function, i.e. pass in the targets to this function, and then the function returns this struct, which we can assert on, but still need to explore how feasible it would be to write tests around this function.

So I wanted to raise this PR and gather feedback if this approach makes sense, before proceeding further.